src/common/internet.rs

```diff
@@ -118,6 +122,7 @@ impl InetAddr {
             InetAddr::IPv6(ipv6_addr) => buf[16..].copy_from_slice(&ipv6_addr.octets()),
             #[cfg(feature = "tor")]
             InetAddr::Tor(tor_addr) => buf = tor_addr.to_bytes(),
+            InetAddr::TorV2(tor_addr) => buf[22..].copy_from_slice(tor_addr.get_raw_bytes().as_ref()),
```

Comment:
Will do.

Reply:
On a second thought, should the line above also be changed to `tor_pubkey`? To make it semantically clear what is the inner structure. 
Maybe keep them as `tor_pubkey` and `tor_addr` respectively? Because the difference between `tor_addr` and `onion_addr` doesn't seem very clear to me.

Reply:
Yes, let's have `tor_pubkey` and `onion_addr`